WooCommerce/Classes/ViewRelated/Coupons/Add and Edit Coupons/AddEditCoupon.swift

```diff
@@ -92,217 +92,223 @@ struct AddEditCoupon: View {
                         }
 
                         LazyNavigationLink(destination: CouponExpiryDateView(date: viewModel.expiryDateField ?? Date(),
-                                                                             isRemovalEnabled: viewModel.expiryDateField != nil,
-                                                                             timezone: viewModel.timezone,
-                                                                             onCompletion: { updatedExpiryDate in
-                            viewModel.expiryDateField = updatedExpiryDate
-                        }), isActive: $showingCouponExpiryDate) {
+                                isRemovalEnabled: viewModel.expiryDateField != nil,
+                                timezone: viewModel.timezone,
+                                onCompletion: { updatedExpiryDate in
+                                    viewModel.expiryDateField = updatedExpiryDate
+                                }), isActive: $showingCouponExpiryDate) {
                             EmptyView()
                         }
 
                         LazyNavigationLink(destination: CouponRestrictions(viewModel: viewModel.couponRestrictionsViewModel),
-                                           isActive: $showingCouponRestrictions) {
+                                isActive: $showingCouponRestrictions) {
                             EmptyView()
                         }
                     }
                 }
             }
             .sheet(isPresented: $showingSelectProducts) {
                 ProductSelector(configuration: ProductSelector.Configuration.productsForCoupons,
-                                isPresented: $showingSelectProducts,
-                                viewModel: viewModel.productSelectorViewModel)
-                    .onDisappear {
-                        viewModel.productSelectorViewModel.clearSearchAndFilters()
-                    }
+                        isPresented: $showingSelectProducts,
+                        viewModel: viewModel.productSelectorViewModel)
+                        .onDisappear {
+                            viewModel.productSelectorViewModel.clearSearchAndFilters()
+                        }
             }
             .sheet(isPresented: $showingSelectCategories) {
                 ProductCategorySelector(isPresented: $showingSelectCategories,
-                                        viewConfig: categorySelectorConfig,
-                                        categoryListConfig: categoryListConfig,
-                                        viewModel: viewModel.categorySelectorViewModel)
+                        viewConfig: categorySelectorConfig,
+                        categoryListConfig: categoryListConfig,
+                        viewModel: viewModel.categorySelectorViewModel)
+            }
+            .sheet(isPresented: $viewModel.showingCouponCreationSuccess) {
+                let couponCode = viewModel.coupon?.code ?? ""
```

Comment:
I'm not sure how we should add the `guard let` here 😅, the `sheet` declaration demands that at least a view is returned, so using the `guard let` triggers this build error here: 
<img width="1012" alt="image" src="https://user-images.githubusercontent.com/5920403/174832825-1d21e327-7090-4925-969b-dec206a27d5c.png">

Since the `showingCouponCreationSuccess` can only be true once a `.success` result is returned, I think that maybe we could move forward displaying the coupon creation success without displaying the coupon code and log the error instead of doing nothing after the creation response is sent to the `AddEditCouponViewModel`, WDYT? I think the creation success view deals nicely with this scenario:

![Simulator Screen Shot - iPhone 13 Pro Max - 2022-06-21 at 12 07 06](https://user-images.githubusercontent.com/5920403/174833861-2deefa20-6d93-49cb-826e-fb6b61472c0f.png)

Reply:
You are right, I missed the fact that we are in a view builder here. Please proceed with your suggestion:

>  I think that maybe we could move forward displaying the coupon creation success without displaying the coupon code and log the error

Thanks!

Reply:
Nice! I've adjusted this part with the error log right here: https://github.com/woocommerce/woocommerce-ios/pull/7052/commits/0b265751b76e92bcc83cc281e41ce666f81d3ce5. But I had to do a hacky solution with the Log call using a wildcard declaration since the ViewBuilder doesn't allow me to declare a log function inside of that code 😅

Reply:
This definitely looks weird, but I cannot come up with a better solution so let's keep it as is 😅